pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -48,3 +77,41 @@ func (pc *PackageControllerClient) ValidateControllerDoesNotExist(ctx context.Co
 	}
 	return nil
 }
+
+func (pc *PackageControllerClient) ApplySecret(ctx context.Context) error {
+	eksaAccessKeyId, eksaSecretAccessKey := os.Getenv(config.EksaAccessKeyIdEnv), os.Getenv(config.EksaSecretAcessKeyEnv)
+	eksaRegion := eksaDefaultRegion
+	if region, found := os.LookupEnv(config.EksaRegionEnv); found {
+		eksaRegion = region
+	}
+
+	templateValues := map[string]string{
+		"eksaAccessKeyId":     base64.StdEncoding.EncodeToString([]byte(eksaAccessKeyId)),
+		"eksaSecretAccessKey": base64.StdEncoding.EncodeToString([]byte(eksaSecretAccessKey)),
+		"eksaRegion":          base64.StdEncoding.EncodeToString([]byte(eksaRegion)),
+	}
+
+	result, err := templater.Execute(awsSecretYaml, templateValues)
+	if err != nil {
+		return fmt.Errorf("replacing template values %v", err)
+	}
+
+	params := []string{"create", "-f", "-", "--kubeconfig", pc.kubeConfig}
+	stdOut, err := pc.kubectl.CreateFromYaml(ctx, result, params...)
```

Comment:
i know this is not part of this PR, but I think we should revisit this method signature. It feels non intuitive that we need to pass "create" in the params when the method is already named create.

Reply:
I will be having a PR to remove feature flag. I'll revisit this when that PR is ready.